File: Hello_World_Tutorials/tutorial-source-code/AVLtree/avl.cpp
#include <string>
#include "Bridges.h"
#include "AVLTreeElement.h"


using namespace std;
using namespace bridges;

int main() {


	Bridges::initialize(11, "YOUR_USER_ID", "YOUR_API_KEY"); 

	AVLTreeElement&lt;int, string&gt; *tle0 = 
				new AVLTreeElement&lt;int, string&gt;(10, "10", "10");
	AVLTreeElement&lt;int, string&gt; *tle1 = 
				new AVLTreeElement&lt;int, string&gt;(5, "5", "5");
	AVLTreeElement&lt;int, string&gt; *tle2 = 
				new AVLTreeElement&lt;int, string&gt;(3, "3", "3");
	AVLTreeElement&lt;int, string&gt; *tle3=  
				new AVLTreeElement&lt;int, string&gt;(20, "20","20");

					 //  link the nodes
	tle0->setLeft(tle1);
	tle0->setRight(tle3);
	tle1->setLeft(tle2);

					// set attributes
	tle0->getVisualizer()->setColor(Color("red"));
	tle1->getVisualizer()->setColor(Color("green"));
	tle2->getVisualizer()->setColor(Color("blue"));
	tle3->getVisualizer()->setColor(Color("yellow"));

	tle0->getLinkVisualizer(tle1)->setColor(Color("red"));
	tle1->getLinkVisualizer(tle2)->setColor(Color("green"));
	tle0->getLinkVisualizer(tle3)->setColor(Color("blue"));

					// illustrate balance factors in the node labels
	tle0->setBalanceFactor(-1);
	tle1->setBalanceFactor(-1);
	tle2->setBalanceFactor(0);
	tle3->setBalanceFactor(0);

					// display the balance factors (on mouseover)
	tle0->setLabel(std::to_string(tle0->getBalanceFactor()));
	tle1->setLabel(std::to_string(tle1->getBalanceFactor()));
	tle2->setLabel(std::to_string(tle2->getBalanceFactor()));
	tle3->setLabel(std::to_string(tle3->getBalanceFactor()));

	Bridges::setTitle("AVL Tree Example");
					// provide BRIDGES the  handle to the tree structure
	Bridges::setDataStructure(tle0);


	Bridges::visualize();

	return 0;
}

File: Hello_World_Tutorials/tutorial-source-code/AVLtree/avl.java
import bridges.connect.Bridges;
import bridges.data_src_dependent.Tweet;
import bridges.base.AVLTreeElement;

public class avl {
    public static void main(String[] args) throws Exception{
        
                    //create the Bridges object
		Bridges bridges = new Bridges(11, "YOUR_API_KEY", "YOUR_USER_ID");

                    //create  an  AVL tree elements
		AVLTreeElement&lt;Integer,String&gt; avl_root = new AVLTreeElement&lt;Integer,String&gt;(50, 
													"Hello");
		AVLTreeElement&lt;Integer,String&gt; avl_2 = new AVLTreeElement&lt;Integer,String&gt;(30, "This");
		AVLTreeElement&lt;Integer,String&gt; avl_3 = new AVLTreeElement&lt;Integer,String&gt;(70, "is");
		AVLTreeElement&lt;Integer,String&gt; avl_4 = new AVLTreeElement&lt;Integer,String&gt;(20, "an");
		AVLTreeElement&lt;Integer,String&gt; avl_5 = new AVLTreeElement&lt;Integer,String&gt;(40, "AVL");
		AVLTreeElement&lt;Integer,String&gt; avl_6 = new AVLTreeElement&lt;Integer,String&gt;(90, "Tree");
                    // put in the balance factors - done manually  here 
                    // for illustration, but in a real project, the user will
                    // implement this as part of the insert/delete
                    // operations on the tree to maintain a balanced tree
		avl_root.setLabel("0");
		avl_2.setLabel("0");
		avl_3.setLabel("-1");
		avl_4.setLabel("0");
		avl_5.setLabel("0");
		avl_6.setLabel("0");

					// add the links between nodes
		avl_root.setLeft(avl_2);
		avl_root.setRight(avl_3);
		avl_2.setLeft(avl_4);
		avl_2.setRight(avl_5);
		avl_3.setRight(avl_6);
                                    // set some attributes
		avl_root.getVisualizer().setColor("magenta");
		avl_root.getVisualizer().setOpacity(0.8f);

									//set visualizer type
		bridges.setDataStructure(avl_root);
				
									// visualize the list
		bridges.visualize();
    }
}
